contracts/interfaces/token/TokenConverter.sol

```diff
@@ -1,21 +1,28 @@
 pragma solidity 0.5.10;
 
-import './Token.sol';
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
 
 
 interface TokenConverter {
 
     function convert(
-        Token _fromToken,
-        Token _toToken,
-        uint256 _fromAmount,
-        uint256 _minReturn
-    ) external payable returns (uint256 amount);
+        IERC20 _token,
+        IERC20 _outToken,
+        uint256 _amount,
+        uint256 _tokenCost,
```

Comment:
Maybe the conversion for buying and selling can be handled in separated methods, something like this.

```
function convertFrom(
    IERC20 _from,
    IERC20 _to,
    uint256 _amount,
    uint256 _minReceive
) external payable returns (uint256 spent);

function convertTo(
    IERC20 _from,
    IERC20 _to,
    uint256 _amount,
    uint256 _maxSpent
) external payable returns (uint256 bought);
```

And similar methods to get returns without doing the convertion.